lua/paramset.lua

```diff
@@ -40,11 +40,43 @@ end
 
 --- add generic parameter
 function ParamSet:add(args)
-  local param = args.param -- param is mandatory
+  local param = args.param
+  if param == nil then
+    if args.number then
+      local id = args.number
+      local name = args.name or id
+      param = number.new(id, name, args.min, args.max, args.default)
```

Comment:
in general, i wonder if, in addition to docs, we don't want to some validation to ensure required args are provided.

if we had something like

```lua
local function checkRequired(args, names) {
  for _,name in names do
    if args.name == nil then error('argument "'..name..'" is required')
  end
} 
```

we could do something like:

```lua
  checkRequired(args, {'min', 'max'});
```

which should report something useful like `argument "max" is required` (only better) if someone did something like:

```lua
params:add{number = 'foo', min = 0}
```

(caveat: all untested; just food for thought!)

Reply:
this sounds nice, certainly. but most fields really aren't required... they just need sensible defaults. have you scanned through to see if there are many required fields?